### PR TITLE
Do not allow an excessive count of an individual event: they indicate a failure

### PR DESCRIPTION
**Pull request by dmage**
Based on @deads2k work #26173.

### CODE REVIEW

File: pkg/synthetictests/events.go [hunk 1]
```diff
@@ -10,6 +10,42 @@ import (
...
+			s += "|"
+		}
+		s += r.String()
+	}
+	return regexp.MustCompile(s)
+}
+
+var knownEvents = combinedRegexp(
+	// [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails [Suite:openshift/conformance/parallel] [Suite:k8s]
+	// PauseNewPods causes readiness probe to fail.
```
Comment (deads2k) [inline pkg/synthetictests/events.go:26]: nit: PauseNewPods intentionally causes

File: pkg/synthetictests/events.go [hunk 2]
```diff
@@ -10,6 +10,42 @@ import (
...
+	regexp.MustCompile(`ns/e2e-statefulset-[0-9]+ pod/ss-[0-9] node/[a-z0-9.-]+ - reason/Unhealthy Readiness probe failed: `),
+
+	// [sig-apps] StatefulSet Basic StatefulSet functionality [StatefulSetBasic] should perform rolling updates and roll backs of template modifications [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
+	regexp.MustCompile(`ns/e2e-statefulset-[0-9]+ pod/ss2-[0-9] node/[a-z0-9.-]+ - reason/Unhealthy Readiness probe failed: HTTP probe failed with statuscode: 404`),
```
Comment (deads2k) [inline pkg/synthetictests/events.go:30]: also intentional?

**Comment by deads2k** [comment]
These look plausible.  if they work, we'll merge.  The approach is very good and should be continued until we're able to merge.  This will be a big help latching our progress.

File: pkg/synthetictests/events.go (hunk 2 above)
Comment (dmage) [reply pkg/synthetictests/events.go:30]: yes, it calls `breakPodHTTPProbe`. Perhaps I need to add a commend about it

**Comment by deads2k** [comment]
/approve

This is good in concept and the ratchet looks solid.  You may end up needing a set of excludes specific to upgrades.

**Comment by dmage** [comment]
e2e-agnostic-cmd:

```
event happened 23 times, something is wrong: ns/cmd-basicresources deploymentconfig/test-deployment-config - reason/DeploymentAwaitingCancellation Deployment of version 2 awaiting cancellation of older running deployments
event happened 21 times, something is wrong: ns/openshift-machine-api machine/ci-op-r7y800m4-098b0-vgskh-worker-westus-sdqlb - reason/Updated Updated machine "ci-op-r7y800m4-098b0-vgskh-worker-westus-sdqlb"
```

Perhaps these can be ignored?

/retest

**Comment by deads2k** [comment]
This looks like a good starting point to me.  If some platforms break, then we can punch out for them using what is developed here or fix them depending on whether the fix is fast.

/lgtm

**Comment by dmage** [comment]
/hold

```
event happened 23 times, something is wrong: ns/openshift-network-diagnostics pod/network-check-target-qxkld node/ip-10-0-138-212.us-west-1.compute.internal - reason/NetworkNotReady network is not ready: container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/kubernetes/cni/net.d/. Has your network provider started?
```

**Pull request merged**